scripts/generate_ssl_server_defaults.cr

```diff
@@ -0,0 +1,46 @@
+#! /usr/bin/env crystal
+#
+# This helper fetches the Mozilla recommendations for default TLS ciphers
+# (https://wiki.mozilla.org/Security/Server_Side_TLS) and automatically places
+# them in src/openssl/ssl/defaults.cr
+
+require "http"
+require "json"
+
+url = ARGV.shift? || "https://ssl-config.mozilla.org/guidelines/latest.json"
+DEFAULTS_FILE = "src/openssl/ssl/defaults.cr"
+
+json = JSON.parse(HTTP::Client.get(url).body)
+
+File.open(DEFAULTS_FILE, "w") do |file|
+  file.print <<-CR
+  # THIS FILE WAS AUTOMATICALLY GENERATED BY script/ssl_server_defaults.cr
+  # on #{Time.utc}.
+
+  abstract class OpenSSL::SSL::Context
+  CR
+
+  configuration = json["configurations"].as_h.each do |level, configuration|
```

Comment:
Could we use JSON::Serializable? I think it would make the code a bit easier to read.

Reply:
I don't think that improves much. Would just remove a few square brackets and as_a.

Reply:
Yeah, maybe it's not worth it, I don't know.

Just in case, here's the mapping if we decide to use it:

```crystal
record Guidelines,
  href : String,
  version : Float64,
  configurations : Hash(String, Configuration) do
  include JSON::Serializable
end

record Configuration,
  oldest_clients : Array(String),
  ciphersuites : Array(String),
  ciphers : Ciphers do
  include JSON::Serializable
end

record Ciphers, openssl : Array(String) do
  include JSON::Serializable
end
```